editor/src/messages/portfolio/document/document_message_handler.rs

```diff
@@ -551,13 +552,40 @@ impl MessageHandler<DocumentMessage, (u64, &InputPreprocessorMessageHandler, &Pe
 					.into(),
 				);
 			}
-			NudgeSelectedLayers { delta_x, delta_y } => {
+			NudgeResizeSelectedLayer { delta_x, delta_y, alt, ctrl } => {
 				self.backup(responses);
+				let keyboard = &ipp.keyboard;
+				let is_scale = keyboard.key(alt);
+				let is_bottom_right = !keyboard.key(ctrl);
 				for path in self.selected_layers().map(|path| path.to_vec()) {
-					let operation = DocumentOperation::TransformLayerInViewport {
-						path,
-						transform: DAffine2::from_translation((delta_x, delta_y).into()).to_cols_array(),
-					};
+					let mut transform_matrix = DAffine2::from_translation((delta_x, delta_y).into()).to_cols_array();
```

Comment:
Mutating this variable can be avoided. And it's inefficient to assume this is a translation before overwriting it with a scale. I'd suggest using the `else` case instead.